pkg/storkctl/schedulepolicy.go

```diff
@@ -60,6 +67,215 @@ func newGetSchedulePolicyCommand(cmdFactory Factory, ioStreams genericclioptions
+				return
+			}
+			schedulePolicyName := args[0]
+			var policyItem storkv1.SchedulePolicyItem
+			schedulePolicyType = strings.ToLower(schedulePolicyType)
+			switch schedulePolicyType {
+			case "interval":
+				var intervalPolicy storkv1.IntervalPolicy
+				intervalPolicy.IntervalMinutes = intervalMinutes
+				if retain == 0 {
+					intervalPolicy.Retain = storkv1.DefaultIntervalPolicyRetain
+				} else {
+					intervalPolicy.Retain = storkv1.Retain(retain)
+				}
+				// Validate the user input
+				err := intervalPolicy.Validate()
+				if err != nil {
+					util.CheckErr(err)
+					return
+				}
+				policyItem.Interval = &intervalPolicy
+			case "daily":
+				var dailyPolicy storkv1.DailyPolicy
+				dailyPolicy.Time = time
+				dailyPolicy.ForceFullSnapshotDay = dailyForceFullSnapshotDay
+				if retain == 0 {
+					dailyPolicy.Retain = storkv1.DefaultDailyPolicyRetain
+				} else {
+					dailyPolicy.Retain = storkv1.Retain(retain)
+				}
+				err := dailyPolicy.Validate()
+				if err != nil {
+					util.CheckErr(err)
+					return
+				}
+				policyItem.Daily = &dailyPolicy
+			case "weekly":
+				var weeklyPolicy storkv1.WeeklyPolicy
+				weeklyPolicy.Time = time
+				weeklyPolicy.Day = dayOfWeek
+				if retain == 0 {
+					weeklyPolicy.Retain = storkv1.DefaultWeeklyPolicyRetain
+				} else {
+					weeklyPolicy.Retain = storkv1.Retain(retain)
+				}
+				err := weeklyPolicy.Validate()
+				if err != nil {
+					util.CheckErr(err)
+					return
+				}
+				policyItem.Weekly = &weeklyPolicy
+			case "monthly":
+				var monthlyPolicy storkv1.MonthlyPolicy
+				monthlyPolicy.Time = time
+				monthlyPolicy.Date = dateOfMonth
+				if retain == 0 {
+					monthlyPolicy.Retain = storkv1.DefaultMonthlyPolicyRetain
+				} else {
+					monthlyPolicy.Retain = storkv1.Retain(retain)
+				}
+				err := monthlyPolicy.Validate()
+				if err != nil {
+					util.CheckErr(err)
+					return
+				}
+				policyItem.Monthly = &monthlyPolicy
+			default:
+				util.CheckErr(fmt.Errorf("need to provide a valid schedule policy type. Valid schedule types are Interval, Daily, Weekly and Monthly"))
+				return
+			}
+			var schedulePolicy storkv1.SchedulePolicy
+			schedulePolicy.Name = schedulePolicyName
+			schedulePolicy.Policy = policyItem
+			_, err := storkops.Instance().CreateSchedulePolicy(&schedulePolicy)
+			if err != nil {
+				util.CheckErr(err)
+				return
+			}
+			msg := fmt.Sprintf("schedule policy %v created successfully", schedulePolicy.Name)
+			printMsg(msg, ioStreams.Out)
+		},
+	}
+	// Picking up user inputs and setting the defaults for flags
+	createSchedulePolicyCommand.Flags().StringVarP(&schedulePolicyType, "policy-type", "t", "Interval", "Select Type of schedule policy to apply. Interval / Daily / Weekly / Monthly.")
+	createSchedulePolicyCommand.Flags().IntVarP(&intervalMinutes, "interval-minutes", "i", 30, "Specify the interval, in minutes, after which Portworx should trigger the operation.")
```

Comment:
Lets avoid using Portworx verbage here and instead say Stork will trigger the operation.

Reply:
Okay. Will address in the follow-up PR

Reply:
Hi @adityadani ,
Since this PR was merged, I have addressed these comments as part of a new improvement PR https://github.com/libopenstorage/stork/pull/1535.
Along with your comments I have also made some changes to the validations during creating schedule policy